releasenotes/notes/add-uncompressed-payload-size-limit-83d46a4ae800a92d.yaml

```
@@ -0,0 +1,13 @@
+# Each section from every releasenote are combined when the
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - |
+    Adds a size limit of 64MB to uncompressed metrics payloads. Uncompressed
+    mertics payloads above this size will be split into smaller payloads before
+    being sent.
```

Note that the "regular" metric series, events and service checks payloads now use a different logic by default to serialize, compress, and check the size of payloads (https://github.com/DataDog/datadog-agent/tree/7.22.x/pkg/serializer/jsonstream, which does check for the uncompressed size as well), so your changes will basically only apply to sketch payloads (and metadata payloads, but those can't be split anyway at the moment).

So we can update this to something like:
```suggestion
    Enforces a size limit of 64MB to uncompressed sketch payloads (distribution metrics).
    Payloads above this size will be split into smaller payloads before being sent.
```